### PR TITLE
BaSP-TG-56: Employees shared components & redux

### PR DESCRIPTION
**Pull request by Javier-Cervera**
Implementation of shared components: Modal Form, Modal Confirm, Modal Success/Error, Buttons (Add / Edit / Delete / Text), Table, Inputs and Select, and Preloader.

Implementation of Redux, with Thunks for fetchs methods: GET all, POST, PUT, and DELETE.

### CODE REVIEW

**Review by agusting19 — APPROVED** [review]
Nice buddy! It's working fine

File: src/Components/Employees/Form/index.jsx [hunk 1]
```diff
@@ -1,204 +1,155 @@
...
-              ? []
-              : employeeInput.projects.toString().replace(/\s+/g, '').split(','),
-          timeSheets:
-            employeeInput.timeSheets.length === 0
-              ? []
-              : employeeInput.timeSheets.toString().replace(/\s+/g, '').split(',')
-        }),
-        headers: {
-          'Content-type': 'application/json'
+      dispatch(updateEmployee(userInput, item._id, (response) => setResponse(response))).then(
```
Comment (NicolasFontana) [inline src/Components/Employees/Form/index.jsx:50]: This is just a comment, this works fine!
Why do you pass this arrow function `(response) => setResponse(response)` as an argument to `createEmployee`, wouldn't it be more readable to pass `setResponse` to it?

`dispatch(createEmployee(userInput, setResponse))`

**Review by NicolasFontana — APPROVED** [review]

**Comment by tinchorova** [comment]
## Testing results

Tested locally on: https://alexis-trackgenix-app-git-feature-tg-56-emplo-95386b-basp-m2022.vercel.app/employees

**Test result: Failed** ⚠️ 

bugs found:
1. Editing without making changes shouldn't be allowed, instead it shows a success message 
image
2. There is a bug with table's and footer's length, creating a blank space and a horizontal scrollbar
image

**Review by tinchorova — CHANGES_REQUESTED** [review]
please refer to: https://github.com/BaSP-m2022/alexis-trackgenix-app/pull/23#issuecomment-1159288223

**Review by tinchorova — APPROVED** [review]
Good job!

**Pull request merged**